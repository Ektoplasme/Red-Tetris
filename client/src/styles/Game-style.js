const blockSize = 40;

export const GameStyle = () => ({
gameInfo: {
    margin: '10px 0px',
    alignItem: 'center',
    // fontWeight: 'bold',
    fontSize: '20px',
    color: 'pink',
    border: '2px solid pink',
    borderRadius: '10px',
    padding: '5px 0px',
    width: '80%'
	},
	finishGameTitle: {
		fontSize: '30px',
		color: 'white',
		fontWeight: 'bold',
		padding: '5px 0px',
		fontFamily: 'Orbitron, sans-serif',
	},
	finishGameInfo: {
		fontSize: '20px',
		color: 'white',
		padding: '5px 0px',
		fontFamily: 'Orbitron, sans-serif',
	},
    gameOverContainer: {
        minHeight: '600px',
        padding: '20px 0px',
        width: '650px',
        borderRadius: '10px',
        backgroundColor: 'pink'
    },
    restartLabel: {
        fontSize: '50px',
        fontWeight: 'bold',
        color: 'white',
        padding: '10px',
        cursor: 'pointer',
        '&:hover': {
            color: 'salmon'
        }
    },
    input:{
        height: '45px',
        width: '250px',
        fontSize: '28px',
        textAlign: 'center',
        backgroundColor: 'rgb(47, 49, 64)',
        border: '3px solid rgb(71, 72, 99)',
        color: 'white',
        fontFamily: 'Orbitron, sans-serif',
        '&:focus': {
          outline: 'none'
        }
      },
    restartButton: {
        cursor: 'pointer',
        '& :active': {
            transform: `scale(0.9, 0.9)`,
        }
    },
    blockStyle: {
        backgroundColor: 'pink',
        width: `${blockSize}px`,
        height: `${blockSize}px`,
        border: '1px solid pink'
    },
    button: {
        width: '100%',
        fontSize: '20px',
        color: 'white',
        backgroundColor: 'pink',
        border: '2px solid pink',
        borderRadius: '10px',
        padding: '10px 0px',
        cursor: 'pointer',
        margin: 5,
        marginLeft: 0,
        '&:hover': {
            backgroundColor: 'inherit',
            color: 'pink'
        }
    }
})