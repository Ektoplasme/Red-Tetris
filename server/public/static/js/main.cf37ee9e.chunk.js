(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{1:function(e){e.exports={I:{color:"red",position:[{rightSpace:0,leftSpace:0,downSpace:2,maxHeight:2,maxWidth:4,width:[0,4,0,0],height:[2,2,2,2],form:[[0,0,0,0],[1,1,1,1],[0,0,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:1,downSpace:0,maxHeight:4,width:[2,2,2,2],maxWidth:2,height:[0,4,0,0],form:[[0,1,0,0],[0,1,0,0],[0,1,0,0],[0,1,0,0]]},{rightSpace:0,leftSpace:0,maxWidth:4,downSpace:1,width:[0,0,4,0],height:[3,3,3,3],maxHeight:3,form:[[0,0,0,0],[0,0,0,0],[1,1,1,1],[0,0,0,0]]},{rightSpace:1,leftSpace:2,maxHeight:4,downSpace:0,width:[3,3,3,3],maxWidth:3,height:[0,0,4,0],form:[[0,0,1,0],[0,0,1,0],[0,0,1,0],[0,0,1,0]]}]},O:{color:"blue",position:[{rightSpace:1,maxHeight:2,leftSpace:1,width:[3,3,0,0],maxWidth:3,downSpace:2,height:[0,2,2,0],form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:1,downSpace:2,width:[3,3,0,0],maxHeight:2,height:[0,2,2,0],form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{height:[0,2,2,0],rightSpace:1,leftSpace:1,width:[3,3,0,0],maxWidth:3,maxHeight:2,downSpace:2,form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]},{height:[0,2,2,0],width:[3,3,0,0],maxWidth:3,rightSpace:1,leftSpace:1,maxHeight:2,downSpace:2,form:[[0,2,2,0],[0,2,2,0],[0,0,0,0],[0,0,0,0]]}]},T:{color:"yellow",position:[{rightSpace:1,leftSpace:0,maxWidth:3,height:[2,2,2,0],width:[2,3,0,0],maxHeight:2,downSpace:2,form:[[0,3,0,0],[3,3,3,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,height:[0,3,2,0],width:[2,3,2,0],maxWidth:3,downSpace:1,maxHeight:3,form:[[0,3,0,0],[0,3,3,0],[0,3,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,height:[2,3,2,0],maxWidth:3,maxHeight:3,width:[0,3,2,0],downSpace:1,form:[[0,0,0,0],[3,3,3,0],[0,3,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,width:[2,3,2,0],maxWidth:3,height:[2,3,0,0],maxHeight:3,downSpace:1,form:[[0,3,0,0],[3,3,0,0],[0,3,0,0],[0,0,0,0]]}]},L:{color:"purple",position:[{rightSpace:1,leftSpace:0,maxHeight:2,width:[3,3,0,0],maxWidth:3,height:[2,2,2,0],downSpace:2,form:[[0,0,4,0],[4,4,4,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,height:[0,3,3,0],maxHeight:3,maxWidth:3,width:[2,2,3,0],downSpace:1,form:[[0,4,0,0],[0,4,0,0],[0,4,4,0],[0,0,0,0]]},{rightSpace:1,maxHeight:3,leftSpace:0,downSpace:1,maxWidth:3,width:[0,3,1,0],height:[3,2,2,0],form:[[0,0,0,0],[4,4,4,0],[4,0,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxHeight:3,maxWidth:2,width:[2,2,2,0],height:[1,3,0,0],downSpace:1,form:[[4,4,0,0],[0,4,0,0],[0,4,0,0],[0,0,0,0]]}]},J:{color:"salmon",position:[{rightSpace:1,leftSpace:0,height:[2,2,2,0],maxHeight:2,width:[1,3,0,0],maxWidth:3,downSpace:2,form:[[5,0,0,0],[5,5,5,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,maxHeight:3,maxWidth:3,height:[0,3,1,0],width:[3,2,2,0],downSpace:1,form:[[0,5,5,0],[0,5,0,0],[0,5,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,width:[0,3,3,0],height:[2,2,3,0],maxWidth:3,maxHeight:3,downSpace:1,form:[[0,0,0,0],[5,5,5,0],[0,0,5,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,maxHeight:3,width:[2,2,2,0],height:[3,3,0,0],downSpace:1,form:[[0,5,0,0],[0,5,0,0],[5,5,0,0],[0,0,0,0]]}]},Z:{color:"cyan",position:[{rightSpace:1,leftSpace:0,height:[1,2,2,0],maxHeight:2,width:[2,3,0,0],maxWidth:3,downSpace:2,form:[[6,6,0,0],[0,6,6,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:1,width:[3,3,2,0],maxHeight:3,height:[0,3,2,0],downSpace:1,form:[[0,0,6,0],[0,6,6,0],[0,6,0,0],[0,0,0,0]]},{rightSpace:1,maxWidth:3,leftSpace:0,maxHeight:3,width:[0,2,3,0],height:[2,3,3,0],downSpace:1,form:[[0,0,0,0],[6,6,0,0],[0,6,6,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,maxHeight:3,width:[2,2,1,0],height:[3,2,0,0],downSpace:1,form:[[0,6,0,0],[6,6,0,0],[6,0,0,0],[0,0,0,0]]}]},S:{color:"green",position:[{rightSpace:1,leftSpace:0,height:[2,2,1,0],width:[3,2,0,0],maxHeight:2,maxWidth:3,downSpace:2,form:[[0,7,7,0],[7,7,0,0],[0,0,0,0],[0,0,0,0]]},{rightSpace:1,leftSpace:1,maxHeight:3,height:[0,2,3,0],width:[2,3,3,0],maxWidth:3,downSpace:1,form:[[0,7,0,0],[0,7,7,0],[0,0,7,0],[0,0,0,0]]},{rightSpace:1,leftSpace:0,maxWidth:3,maxHeight:3,width:[0,3,2,0],height:[3,3,2,0],downSpace:1,form:[[0,0,0,0],[0,7,7,0],[7,7,0,0],[0,0,0,0]]},{rightSpace:2,leftSpace:0,maxWidth:2,width:[1,1,2,0],maxHeight:3,height:[2,3,0,0],downSpace:1,form:[[7,0,0,0],[7,7,0,0],[0,7,0,0],[0,0,0,0]]}]}}},13:function(e,t,a){e.exports=a(25)},18:function(e,t,a){},19:function(e,t,a){},20:function(e,t,a){},25:function(e,t,a){"use strict";a.r(t);var i=a(0),n=a.n(i),o=a(8),r=a.n(o),c=(a(18),a(19),a(20),a(2)),h=a(9),p=a(4),l=a(1),m=a(34);function d(e,t){var a=Object.keys(e);if(Object.getOwnPropertySymbols){var i=Object.getOwnPropertySymbols(e);t&&(i=i.filter(function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable})),a.push.apply(a,i)}return a}function f(e){for(var t=1;t<arguments.length;t++){var a=null!=arguments[t]?arguments[t]:{};t%2?d(a,!0).forEach(function(t){Object(h.a)(e,t,a[t])}):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(a)):d(a).forEach(function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(a,t))})}return e}var s={backgroundColor:"pink",width:"".concat(40,"px"),height:"".concat(40,"px"),border:"1px solid pink"},S=function(e){var t=e.empty,a=e.color,i=e.transparent;return n.a.createElement("div",{style:t?f({},s):f({},s,i?{border:"1px solid transparent",backgroundColor:"transparent"}:{border:"1px solid white",backgroundColor:"".concat(a)})})},u=Object(m.a)(function(){return{gameOverContainer:{height:"500px",width:"500px",backgroundColor:"pink"},scoreLabel:{fontSize:"40px",fontWeight:"bold",color:"pink",marginTop:"20px"},restartLabel:{fontSize:"50px",fontWeight:"bold",color:"white",padding:"10px","&:hover":{color:"salmon"}},restartButton:{cursor:"pointer","& :active":{transform:"scale(0.9, 0.9)"}},blockStyle:{backgroundColor:"pink",width:"".concat(40,"px"),height:"".concat(40,"px"),border:"1px solid pink"}}})(function(e){var t,a=function(e){var t=p.pieces[e],a={maxWidth:l[t].position[0].maxWidth,maxHeight:l[t].position[0].maxHeight,h:l[t].position[0].height,type:l[t].type,name:t,width:l[t].position[0].width,x:3,y:0,leftSpace:l[t].position[0].leftSpace,rightSpace:l[t].position[0].rightSpace,downSpace:l[t].position[0].downSpace,rot:0,form:l[t].position[0].form,color:l[t].color};return!!o(a)&&a},o=function(e){var t=W.tab,a=!0;return e.form.map(function(i,n){i.map(function(i,o){i>0&&(!t[e.y+n]||t[e.y+n][e.x+o]>0||e.x+o>9+e.rightSpace||e.x+o<0-e.leftSpace||"undefined"===typeof t[e.y+n][e.x+o])&&(a=!1)})}),a},r=e.classes,h=Object(i.useState)(0),m=Object(c.a)(h,2),d=m[0],s=m[1],u=Object(i.useState)(!1),g=Object(c.a)(u,2),x=g[0],w=g[1],y=Object(i.useState)(0),O=Object(c.a)(y,2),b=O[0],v=O[1],E=Object(i.useState)({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]}),L=Object(c.a)(E,2),W=L[0],k=L[1],I=Object(i.useState)(a(0)),j=Object(c.a)(I,2),J=j[0],T=j[1],H=Object(i.useState)(0),Z=Object(c.a)(H,2),C=Z[0],N=(Z[1],Object(i.useState)(0)),P=Object(c.a)(N,4),z=P[0],R=P[1],B=P[2],D=(P[3],Object(i.useState)(!0)),A=Object(c.a)(D,2),$=A[0],q=A[1],F=Object(i.useState)(l[p.pieces[1]]),G=Object(c.a)(F,2),K=G[0],M=G[1],Q=Object(i.useRef)(!1);Object(i.useEffect)(function(){JSON.stringify(W)===JSON.stringify({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]})&&d>0&&(s(0),v(0),a(0),T(a(0)),M(l[p.pieces[1]]),w(!1))},[W,d]);var U=function(e){(!t||38!=t.keyCode&&32!=t.keyCode&&40!=t.keyCode)&&(t=e,39===e.keyCode&&$&&T(function(e){return o(f({},e,{x:e.x+1}))?f({},e,{x:e.x+1}):e}),37===e.keyCode&&$&&T(function(e){return o(f({},e,{x:e.x-1}))?f({},e,{x:e.x-1}):e}),32===e.keyCode&&$&&(q(!1),X("hard drop",z-J.y),clearInterval(Q.current),Q.current=V(15)),38===e.keyCode&&$&&T(function(e){var t=e.rot>2?0:e.rot+1,a=f({},e,{form:l[e.name].position[t].form,rot:t,h:l[e.name].position[t].height,leftSpace:l[e.name].position[t].leftSpace,rightSpace:l[e.name].position[t].rightSpace,downSpace:l[e.name].position[t].downSpace,maxHeight:l[e.name].position[t].maxHeight,maxWidth:l[e.name].position[t].maxWidth,width:l[e.name].position[t].width,height:l[e.name].position[t].height});return o(a)?a:e}),40===e.keyCode&&(clearInterval(Q.current),Q.current=V(30)))},V=function(e){return setInterval(function(){T(function(e){if(o(f({},e,{y:e.y+1})))return f({},e,{y:e.y+1});R(e.y+1),q(!0),t=null,clearInterval(Q.current),k(function(t){return e.form.map(function(a,i){a.map(function(a,n){e.form[i][n]>0&&(t.tab[e.y+i][e.x+n]=e.form[i][n])})}),t});var i=a(d+1);return s(function(e){return M(l[p.pieces[e+2]]),e+1}),!1!==i&&o(i)?i:(w(!0),e)})},e)},X=function(e,t){var a=0;switch(e){case"row":a=[40,100,300,1200][t+1];case"hard drop":a=t+1;case"soft drop":a=t}v(function(e){return e+a})};Object(i.useEffect)(function(){return Q.current=V(1e3),x&&clearInterval(Q.current),function(){clearInterval(Q.current)}},[d,x]),Object(i.useEffect)(function(){return document.addEventListener("keydown",U),document.addEventListener("keyup",function(e){40===e.keyCode&&(clearInterval(Q.current),Q.current=V(1e3)),32!==t&&(t=null)}),x&&document.removeEventListener("keydown",U),function(){document.removeEventListener("keydown",U)}},[d,x,$,J,C,z,B]);var Y=["red","blue","yellow","purple","salmon","cyan","green"],_=function(){var e=W.tab,t=[];e.map(function(a,i){return a.every(function(e){return e>0})&&t.push(i),e}),t.length&&(function(e,t){var a=t;e.map(function(t){return a.splice(t,1),a.splice(0,0,[0,0,0,0,0,0,0,0,0,0]),e}),k({tetriList:W.tetriList,tab:a})}(t,e),X("row",t.length))};Object(i.useEffect)(function(){_()});var ee=J.form;return x?n.a.createElement("div",{className:"flex column center alignCenter",style:{height:"100vh"}},n.a.createElement("div",{className:"flex column center alignCenter ".concat(r.gameOverContainer)},n.a.createElement("div",{style:{fontSize:"60px",fontWeight:"bold",color:"salmon"}},"Sorry, you lose"),n.a.createElement("div",{style:{fontSize:"50px",fontWeight:"bold",color:"white"}},"SCORE: ",b),n.a.createElement("div",{className:"flex column center alignCenter ".concat(r.restartButton)},n.a.createElement("div",{className:r.restartLabel,onClick:function(){k({tab:[[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0],[0,0,0,0,0,0,0,0,0,0]],tetriList:[]})}},"RESTART")))):n.a.createElement("div",{className:"flex"},n.a.createElement("div",{className:"absolute",style:{height:"500px",left:"100px"}},n.a.createElement("div",{className:r.scoreLabel},"SCORE: ",b)),n.a.createElement("div",{className:"absolute",style:{top:"100px",left:"100px"}},W.tab.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:Y[e-1]}):n.a.createElement(S,{empty:!0}))}))}),n.a.createElement("div",{className:"absolute",style:{top:"".concat(40*J.y+2*J.y,"px"),left:"".concat(40*J.x+2*J.x,"px")}},ee.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:J.color}):n.a.createElement(S,{transparent:!0}))}))}))),n.a.createElement("div",{className:"absolute flex center alignCenter column",style:{backgroundColor:"lightblue",left:"600px",top:"100px"}},n.a.createElement("div",{style:{fontSize:"40px",fontWeight:"bold",color:"navy",marginTop:"20px"}},"Next:"),n.a.createElement("div",{className:"relative",style:{top:"10px",padding:"10px"}},K.position[0].form.map(function(e,t){return n.a.createElement("div",{style:{display:"flex"},key:t},e.map(function(e,t){return n.a.createElement("div",{key:t},e>0?n.a.createElement(S,{color:K.color}):n.a.createElement(S,{transparent:!0}))}))}))))});Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));r.a.render(n.a.createElement(u,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then(function(e){e.unregister()})},4:function(e){e.exports={pieces:["T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I","T","S","J","Z","L","O","I"]}}},[[13,1,2]]]);
//# sourceMappingURL=main.cf37ee9e.chunk.js.map