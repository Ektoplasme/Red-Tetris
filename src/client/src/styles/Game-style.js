export const blockSize = 40;

export const GameStyle = () => ({
gameInfo: {
    margin: '10px 0px',
    alignItem: 'center',
    // fontWeight: 'bold',
    fontSize: '25px',
    color: 'pink',
    border: '2px solid pink',
    borderRadius: '10px',
    padding: '5px 0px',
    width: '80%'
    },
    scoreContainer: {
        border: '1px solid white',
        padding: 5,
        borderRadius: '10px',
        minWidth: '250px',
        marginTop: '15px',
    },
    winLabel: {
        color: '#d8efbb',
        fontSize: '55px',
        fontWeight: 'bold',
        animation: 'animate 3s infinite'
    },
    finishGameRestart: {
        fontSize: '35px',
		color: 'white',
		fontWeight: 'bold',
		padding: '5px 0px',
		fontFamily: 'Orbitron, sans-serif',
        marginTop: '5px'
    },
	finishGameTitle: {
		fontSize: '50px',
		color: 'white',
		fontWeight: 'bold',
		padding: '5px 0px',
		fontFamily: 'Orbitron, sans-serif',
	},
	finishGameInfo: {
		fontSize: '20px',
		color: 'white',
		padding: '5px 0px',
		fontFamily: 'Orbitron, sans-serif',
	},
    gameOverContainer: {
        minHeight: '600px',
        padding: '20px 0px',
        width: '650px',
        borderRadius: '10px',
        backgroundColor: 'pink'
    },
    restartLabel: {
        fontSize: '35px',
        fontWeight: 'bold',
        color: 'white',
        padding: '10px',
        cursor: 'pointer',
        '&:hover': {
            color: 'salmon'
        }
    },
    input:{
        height: '45px',
        width: '250px',
        fontSize: '28px',
        textAlign: 'center',
        backgroundColor: 'rgb(47, 49, 64)',
        border: '3px solid rgb(71, 72, 99)',
        color: 'white',
        fontFamily: 'Orbitron, sans-serif',
        '&:focus': {
          outline: 'none'
        }
      },
    restartButton: {
        cursor: 'pointer',
        '& :active': {
            transform: `scale(0.9, 0.9)`,
        }
    },
    blockStyle: {
        backgroundColor: 'pink',
        width: `${blockSize}px`,
        height: `${blockSize}px`,
        border: '1px solid pink'
    },
    button: {
        width: '100%',
        fontSize: '20px',
        color: 'white',
        backgroundColor: 'pink',
        border: '2px solid pink',
        borderRadius: '10px',
        padding: '10px 0px',
        cursor: 'pointer',
        margin: 5,
        marginLeft: 0,
        '&:hover': {
            backgroundColor: 'inherit',
            color: 'pink'
        }
    }
})