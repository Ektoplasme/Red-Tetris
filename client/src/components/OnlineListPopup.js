import React from 'react'
import { withStyles }  from '@material-ui/styles'

const styles = () => ({
    back:{
        backgroundColor: 'rgba(33,35,46,0.8)'
    },
    container: {
        width: '300px',
        minHeight: '200px',
        borderRadius: '20px',
        boxShadow: '4px 2px 19px 10px rgba(112,106,112,0.29)',
        backgroundColor: 'rgba(33,35,46,1)',
    },
    title: {
        fontSize: '30px',
        color: 'white',
        padding: 5
    },
    description: {
        fontSize: '18px',
        color: 'white',
        padding: 5
    },
    button: {
        cursor: 'pointer',
        '&:hover': {
            backgroundColor: 'rgba(83,85,96,1)',
            color: 'rgba(43,45,56,1)'
        },
        width: '150px',
        height: '30px',
        borderRadius: '5px',
        backgroundColor: 'rgba(43,45,56,1)',
        color: 'white',
        marginTop: 10
    }
})

const Popup = ({classes, users, close}) => {
		console.log(users)
    return (
        <div style={{zIndex: 100}} className={`${classes.back} fullHeight absolute fullWidth flex center alignCenter`}>
            <div style={{padding: '10px'}} className={`flex center column  alignCenter ${classes.container}`}>
                {/* <div className={classes.title}>
                  Users Online
                </div> */}
                <div style={{minHeight: '100px', width: '200px'}} className={classes.description}>
									{users
										? Object.values(users).map((user, id)=>{
											return <div style={{position: 'relative', left: '20px'}} key={id} className='flex row alignCenter'>
													<div style={{color: 'green', padding: '0px 5px'}}>â€¢</div>
													<div >{user}</div>
												</div>
										})
										: null
									}
                </div>
                <button className={`flex center alignCenter ${classes.button}`} onClick={()=>{close(false)}}>
                  OK
                </button>
            </div>
        </div>
    )
}

export default withStyles(styles)(Popup)