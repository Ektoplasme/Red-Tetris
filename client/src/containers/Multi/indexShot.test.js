import React from 'react';
import { shallow } from 'enzyme';
import styled from 'styled-components';
import renderer from 'react-test-renderer';

import { styles } from '../../components/OnlineListPopup.js';
import { Multi } from './index.js';
import { GameStyle } from '../../styles/Game-style.js';

describe("index.js SnapShot test",() => {
	const styleTmp = styles();
	const gameState = {
	room: null,
	nbPlayer: null,
	player: "NobilaTest",
	playerId: "v34h9492wvkbh3wg3q",
	isInGame: true,
	gameStarted: false,
	isHost: true,
	playTab: [],
	shadows: null,
	onlineUsers: "Josie",
	endOfGame: false,
	winScore: null,
	playing: false,
	pieces: null
};
	test("Menu component Match SnapShot", () => {
		const blockCmp = renderer.create(<Multi classes={ GameStyle } gameState={ gameState } popupInfo={ styles }/>).toJSON();
		expect(blockCmp).toMatchSnapshot();
	});
});
