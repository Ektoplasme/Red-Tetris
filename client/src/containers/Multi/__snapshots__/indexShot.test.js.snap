// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js SnapShot test Menu component Match SnapShot 1`] = `
<div
  className="fullWidth fullHeight"
>
  <div>
    <div
      className="undefined navigationBar fullWidth flex center alignCenter"
      onClick={[Function]}
    >
      <p>
        RETURN MENU
      </p>
    </div>
    <div
      className="App"
      style={
        Object {
          "alignItems": "center",
          "animation": "gradientBackground 20s infinite",
          "backgroundColor": "rgb(33, 35, 46)",
          "backgroundSize": "100% 100%",
          "color": "white",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": "calc(10px + 2vmin)",
          "height": "100%",
          "overflow": "hidden",
        }
      }
    >
      <p
        align="center"
      >
        Waiting for host to start the game
      </p>
      <div
        className="react-dots-loader"
        color={null}
        size={null}
        style={
          Object {
            "display": "inline-block",
          }
        }
      >
        <div
          className="  react-dots-loader__circle  react-dots-loader__circle--one"
          style={
            Object {
              "background": "white",
              "display": "inline-block",
              "height": 18,
              "marginLeft": 0,
              "width": 18,
            }
          }
        />
        <div
          className="  react-dots-loader__circle  react-dots-loader__circle--two"
          style={
            Object {
              "background": "white",
              "display": "inline-block",
              "height": 18,
              "marginLeft": 10,
              "width": 18,
            }
          }
        />
        <div
          className="  react-dots-loader__circle  react-dots-loader__circle--three"
          style={
            Object {
              "background": "white",
              "display": "inline-block",
              "height": 18,
              "marginLeft": 10,
              "width": 18,
            }
          }
        />
      </div>
      <p>
        Room : 
      </p>
      <div>
        <div>
          Players in this room :
        </div>
      </div>
    </div>
  </div>
</div>
`;
